File: LeetCode/0013. Roman to Integer/JavaScript/01. Roman to Integer.js
var romanToInt = function (s) {
    let romanMaps = new Map();
    romanMaps.set("I", 1);
    romanMaps.set("V", 5);
    romanMaps.set("X", 10);
    romanMaps.set("L", 50);
    romanMaps.set("C", 100);
    romanMaps.set("D", 500);
    romanMaps.set("M", 1000);

    let result = 0;
    for (let x = 0; x < s.length; x++) {
        if (x === (s.length - 1)) {
            result += romanMaps.get(s[x].toString());
            break;
        }
        if (romanMaps.get(s[x].toString()) >= romanMaps.get(s[x + 1].toString())) {
            result += romanMaps.get(s[x].toString());
        }
        else {
            result -= romanMaps.get(s[x].toString());
        }
    }
    return result;
}

File: LeetCode/0013. Roman to Integer/C#/01. Roman to Integer.cs
int RomanToInt(string s)
{
    Dictionary<string, int> romanMap = new Dictionary<string, int>
    {
        { "I", 1 },
        { "V", 5 },
        { "X", 10 },
        { "L", 50 },
        { "C", 100 },
        { "D", 500 },
        { "M", 1000 }
    };
    int result = 0;

    for (int x = 0; x < s.Length; x++)
    {
        if (x == s.Length - 1)
        {
            result += romanMap[s[x].ToString()];
            break;
        }
        if (romanMap[s[x].ToString()] >= romanMap[s[x + 1].ToString()])
            result += romanMap[s[x].ToString()];
        else
            result -= romanMap[s[x].ToString()];
    }
    return result;
}